components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/OAuthEventInterceptorListener.java

```diff
@@ -0,0 +1,182 @@
+            IdentityOAuth2Exception {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Listening to the token renewal event");
+        }
+        String tokenId = tokenRespDTO.getTokenId();
+        if (StringUtils.isBlank(tokenId)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Could not find the tokenId of the application: " + tokenReqDTO.getClientId());
+            }
+            return;
+        }
+        String accessToken = tokenRespDTO.getAccessToken();
+        if (StringUtils.isBlank(accessToken)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Since the accesstoken is invalid, couldn't find session context identifier " +
+                        "for the application: " + tokenReqDTO.getClientId());
+            }
+            return;
+        }
+        String sessionContextId = getSessionContextIdentifier(accessToken);
+        if (StringUtils.isBlank(sessionContextId)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Could not find the session context identifier for the application: " +
+                        tokenReqDTO.getClientId());
+            }
+            return;
+        }
+        String tenantDomain = tokenReqDTO.getTenantDomain();
+        if (StringUtils.isBlank(tenantDomain)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Could not find the tenant domain of the application: " + tokenReqDTO.getClientId());
+            }
+            return;
+        }
+        int tenantId = OAuth2Util.getTenantId(tenantDomain);
+        persistTokenToSessionMapping(sessionContextId, tokenId, tenantId);
+    }
+
+    @Override
+    public void onPostTokenIssue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx, AccessTokenDO tokenDO,
+                                 OAuth2AuthorizeRespDTO respDTO, Map<String, Object> params)
+            throws IdentityOAuth2Exception {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Listening to the token renewal event");
+        }
+        String sessionContextId = oauthAuthzMsgCtx.getAuthorizationReqDTO().getIdpSessionIdentifier();
+        if (StringUtils.isBlank(sessionContextId)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Could not find the session context identifier for the application: " +
+                        tokenDO.getConsumerKey());
+            }
+            return;
+        }
+        String tokenId = tokenDO.getTokenId();
+        if (StringUtils.isBlank(tokenId)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Could not find the tokenId of the application: " + tokenDO.getConsumerKey());
+            }
+            return;
+        }
+        int tenantId = tokenDO.getTenantID();
+        persistTokenToSessionMapping(sessionContextId, tokenId, tenantId);
+    }
+
+    public boolean isEnabled() {
+        return true;
```

Comment:
read it from config in identity.xml to disable or enable it

Reply:
if it is not added, it should be enabled. 
if it is added and enabled ->. enabled
if it is added and property ->. enabled